ros_buildfarm/common.py

```diff
@@ -172,7 +172,10 @@ def get_distribution_repository_keys(urls, key_files):
 def get_binary_package_versions(apt_cache, debian_pkg_names):
     versions = {}
     for debian_pkg_name in debian_pkg_names:
-        pkg = apt_cache[debian_pkg_name]
+        pkg = apt_cache.get(debian_pkg_name)
+        if not pkg:
+            prov = apt_cache.get_providing_packages(debian_pkg_name)
+            pkg = apt_cache[prov[0]]  # Just grab the first one
```

Comment:
Is it possible for the Providing package to be absent from the cache as well? I think the answer is no, but if not we probably need to loop the same handling as for a miss on the top-level cache.